packages/main/src/ToolbarButton.ts

```diff
@@ -156,6 +156,13 @@ class ToolbarButton extends ToolbarItem {
 	@property()
 	width?: string;
 
+	onEnterDOM() {
+		const toolbar = this.parentElement as HTMLElement;
+		this._getRealDomRef = () => {
```

Comment:
It is not best practice to look for parent element and the context where the element is placed.

Could you please check how it is fixed in shellbar: https://github.com/SAP/ui5-webcomponents/blob/main/packages/fiori/src/ShellBar.ts#L901

Reply:
Thank you for the feedback! You’re absolutely right; relying on the parent element and its context isn’t ideal.